src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -563,6 +563,9 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
+            
+            if (FileSystem.Current.DirectoryExists(fulldestDirName))
+                throw new IOException(SR.Format(SR.IO_AlreadyExists_Name, fulldestDirName));
```

Comment:
In general, the Windows IO API isn't atomic- the [abandoned](https://msdn.microsoft.com/en-us/library/windows/desktop/hh802690(v=vs.85).aspx) TxF API was introduced in an attempt to make it so.

I looked at the code for a bit and I don't see anything special being done- that said I don't know what difference it makes about how close it gets to the actual attempt to create the destination handle. I'm trying to imagine how the timing could potentially be bad from our API perspective- I'm not coming up with anything worse than the normal "anything can happen at any time". I could be missing something, of course. :)